web/src/components/navigation/navbar.js

```diff
@@ -1,44 +1,69 @@
-import {
-  isWidthUp,
-  withWidth,
-} from '@material-ui/core';
-import {gql} from 'apollo-boost';
-import NavBarMobile from './navbarmobile';
-import NavBarPC from './navbarpc';
-import {Query} from 'react-apollo';
-import React from 'react';
-
-const ME_QUERY = gql`
-query getMyProfilePic {
-  me {
-    profileImgUrl
-  }
-}
-`;
-
-function Navbar(props) {
-  if (isWidthUp('md', props.width)) {
-    return (
-      <Query query={ME_QUERY}>
-        {({loading, error, data}) => {
-          if (loading || error) return <NavBarPC/>;
-          return (
-            <NavBarPC profilepic={data.me.profileImgUrl}/>
-          );
-        }}
-      </Query>
-    );
-  }
-  return (
-    <Query query={ME_QUERY}>
-      {({loading, error, data}) => {
-        if (loading || error) return <NavBarMobile/>;
-        return (
-          <NavBarMobile profilepic={data.me.profileImgUrl}/>
-        );
-      }}
-    </Query>
-  );
-}
-
-export default withWidth()(Navbar);
+import Cookies from 'js-cookie';
+import {
+  isWidthUp,
+  withWidth,
+} from '@material-ui/core';
+import {gql} from 'apollo-boost';
+import NavBarMobile from './navbarmobile';
+import NavBarPC from './navbarpc';
+import {Query} from 'react-apollo';
+import React from 'react';
+
+const ME_QUERY = gql`
+query getMyProfilePic {
+  me {
+    profileImgUrl
+  }
+}
+`;
+
+function Navbar(props) {
+  if (isWidthUp('md', props.width)) {
+    // When there is token, send me query to get profile pic.
+    if (Cookies.get('token')) {
```

Comment:
This logic is duplicated below. Can we try to merge them and use an if block to decide whether to return NavBarPC vs NavBarMobile?